src/ds/d500/hw_monitor_extended_buffers.cpp

```diff
@@ -7,6 +7,20 @@
 
 namespace librealsense
 {
+    std::string hw_monitor_extended_buffers::get_firmware_version_string(const std::vector<uint8_t>& buff, size_t index, size_t length) const
+    {
+        std::stringstream formattedBuffer;
```

Comment:
We need to add some comment here, explaining what's different than the base FW version, how the HKR FW version is, what the schema for it is, etc.

Reply:
I am concerned that, with the higher bit sizes, the numbering is going to be unreadable. These FW versions are visible to the user.

More importantly, we use a common class `rsutils::version`, to later parse version strings and interpret them. I don't like that they've deviated from what we've defined as a schema (1-2-1-4 bytes for major-minor-patch-build). It's dangerous.

So please include an explanation of how the FW versioning is going to work, with an example.

@Nir-Az @ev-mp I think FW versioning should be under our control as it's user-facing. We can use their numbering internally (just like the module version and others) but the public FW version needs to be ours.

Reply:
![image](https://github.com/IntelRealSense/librealsense/assets/64067618/781631c0-8534-41ee-a91f-07941c72ef4f)

Reply:
About being unreadable I am sure Remi will test that, there shouldn't be a reason that it will be unreadable no matter with which field size it will be sent.
We cannot demand each SW component to align to LibRS format.
Are you saying that if FW team will increase to 10.0.0.1 we have an issue? if yes, the problem is on own scheme.